src/HealthChecks.EventStore/EventStoreHealthCheck.cs

```diff
@@ -10,45 +10,46 @@ public class EventStoreHealthCheck : IHealthCheck
         private const int ELAPSED_DELAY_MILLISECONDS = 500;
         private const int RECONNECTION_LIMIT = 1;
 
-        private readonly string _eventStoreConnection;
+        private readonly IEventStoreConnection? _eventStoreConnection;
+        private readonly string? _eventStoreConnectionString;
         private readonly string? _login;
         private readonly string? _password;
 
         public EventStoreHealthCheck(string eventStoreConnection, string? login, string? password)
         {
-            _eventStoreConnection = Guard.ThrowIfNull(eventStoreConnection);
+            _eventStoreConnectionString = Guard.ThrowIfNull(eventStoreConnection);
             _login = login;
             _password = password;
         }
 
+        public EventStoreHealthCheck(IEventStoreConnection eventStoreConnection)
+        {
+            _eventStoreConnection = eventStoreConnection ?? throw new ArgumentNullException(nameof(eventStoreConnection));
+        }
+
         public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
         {
             try
             {
-                ConnectionSettingsBuilder connectionSettings;
+                var connectionSettingsBuilder = ConnectionSettings.Create()
+                    .LimitReconnectionsTo(RECONNECTION_LIMIT)
+                    .SetReconnectionDelayTo(TimeSpan.FromMilliseconds(ELAPSED_DELAY_MILLISECONDS))
+                    .SetCompatibilityMode("auto");
 
-                if (string.IsNullOrEmpty(_login) || string.IsNullOrEmpty(_password))
+                if (!string.IsNullOrEmpty(_login) && !string.IsNullOrEmpty(_password))
                 {
-                    connectionSettings = ConnectionSettings.Create()
-                        .LimitReconnectionsTo(RECONNECTION_LIMIT)
-                        .SetReconnectionDelayTo(TimeSpan.FromMilliseconds(ELAPSED_DELAY_MILLISECONDS))
-                        .SetCompatibilityMode("auto");
-                }
-                else
-                {
-                    connectionSettings = ConnectionSettings.Create()
-                        .LimitReconnectionsTo(RECONNECTION_LIMIT)
-                        .SetReconnectionDelayTo(TimeSpan.FromMilliseconds(ELAPSED_DELAY_MILLISECONDS))
-                        .SetDefaultUserCredentials(new UserCredentials(_login, _password))
-                        .SetCompatibilityMode("auto");
+                    connectionSettingsBuilder.SetDefaultUserCredentials(new UserCredentials(_login, _password));
                 }
 
-                using (var connection = EventStoreConnection.Create(
-                    _eventStoreConnection,
-                    connectionSettings,
-                    CONNECTION_NAME))
+                using (var connection = _eventStoreConnection ??

```

Comment:
So you dispose `_eventStoreConnection` just after the first call. What about next call on disposed connection?

Reply:
Ok

Reply:
Done